### PR TITLE
Redirect to user profile page after login and try to get user data

### PR DESCRIPTION
**Pull request by Sosodope**

### CODE REVIEW

File: client/src/components/LoginForm/LoginFormContainer.js [hunk 1]
```diff
@@ -37,54 +38,58 @@ const LoginValidation = yup.object().shape({
...
-            }}
-            onSubmit={async (values, { setSubmitting }) => {
-              const response = await loginUser({
-                variables: { email: values.email, password: values.password }
-              });
-              // This is where you could send the submitted values to the backend
-              localStorage.setItem("token", response.data.loginUser.token);
+    render() {
+        const { classes, history } = this.props;
+        console.log(this.props)
```
Comment (ivanms1) [inline client/src/components/LoginForm/LoginFormContainer.js]: forgot a console.log

File: client/src/components/UserProfile/UserBio.js [hunk 2]
```diff
@@ -1,32 +1,31 @@
...
+import dayjs from 'dayjs'
 
 const useStyles = makeStyles(theme => ({
   button: {
     marginRight: '1em',
   }
 }));
 
-const UserBio = () => {
+const UserBio = (props) => {
```
Comment (ivanms1) [inline client/src/components/UserProfile/UserBio.js:12]:
```suggestion
const UserBio = ({ bio }) => {
```

File: client/src/components/UserProfile/UserBio.js [hunk 3]
```diff
@@ -1,32 +1,31 @@
...
 	const classes = useStyles();
-	// const { firstname, lastname, location, createdAt } = this.PaymentResponse;
-	// const fullname = `${firstname} ${lastname}`
+	const { firstName, lastName, location, createdAt, bio } = props.bio;
```
Comment (ivanms1) [inline client/src/components/UserProfile/UserBio.js:14]: inside props or props.bio?

File: client/src/components/UserProfile/UserGroups.js [hunk 4]
```diff
@@ -1,6 +1,9 @@
 import React from 'react'
 
-const UserGroups = () => {
+const UserGroups = (props) => {
+	// TODO:: Query groups here and get group info
+	const { groups } = props
+	console.log(groups)
```
Comment (ivanms1) [inline client/src/components/UserProfile/UserGroups.js]: console.log

File: client/src/components/UserProfile/UserPreferences.js [hunk 5]
```diff
@@ -1,17 +1,23 @@
 import React from 'react'
 
-const UserPreferences = () => {
+const UserPreferences = (props) => {
```
Comment (ivanms1) [inline client/src/components/UserProfile/UserPreferences.js]:
```suggestion
const UserPreferences = ({ categories }) => {
```

File: client/src/components/UserProfile/UserProfileContainer.js [hunk 6]
```diff
@@ -2,14 +2,43 @@ import React from 'react'
...
+								<UserBioContainer userBio={userBio}/>
+						)
+					}}
+				</Query>
+				<Query query={GET_USER_PREFS} variables={{ id }}>
+						{({ data, loading, error }) => {
+							if (loading) return <p>Loading</p>;
+							if (error) return <p>ERROR</p>;
+							const userPrefs = data.getUser
+							console.log(userPrefs)
```
Comment (ivanms1) [inline client/src/components/UserProfile/UserProfileContainer.js]: console.logs

File: client/src/components/UserProfile/UserProfileContainer.js [hunk 7]
```diff
@@ -2,14 +2,43 @@ import React from 'react'
...
+							if (loading) return <p>Loading</p>;
+							if (error) return <p>ERROR</p>;
+							const userBio = data.getUser
+							console.log(userBio)
+							return (
+								<UserBioContainer userBio={userBio}/>
+						)
+					}}
+				</Query>
+				<Query query={GET_USER_PREFS} variables={{ id }}>
```
Comment (ivanms1) [inline client/src/components/UserProfile/UserProfileContainer.js:26]: why not use just one query?

File: client/src/components/UserProfile/queryGetUserInterests.graphql [hunk 8]
```diff
@@ -0,0 +1,9 @@
+query GetUserBio($id: ID!){
```
Comment (ivanms1) [inline client/src/components/UserProfile/queryGetUserInterests.graphql]: this query is not fetching the bio, you can change the name to reflect what the query is actually doing

File: client/src/components/UserProfile/queryGetUserPrefs.graphql [hunk 9]
```diff
@@ -0,0 +1,9 @@
+query GetUserBio($id: ID!){
```
Comment (ivanms1) [inline client/src/components/UserProfile/queryGetUserPrefs.graphql]: same as above

File: client/src/components/UserProfile/UserBio.js (hunk 3 above)
Comment (Sosodope) [reply client/src/components/UserProfile/UserBio.js:14]: errors with just props or if i destructure it

File: client/src/components/UserProfile/UserProfileContainer.js (hunk 7 above)
Comment (Sosodope) [reply client/src/components/UserProfile/UserProfileContainer.js:26]: I was trying to use the same component logic while making my queries so each query returns a single request. Let me know if it's not a suitable approach for this, can change it

**Review by ivanms1 — APPROVED** [review]

**Pull request merged**